The name defined in embed.fnc is `new_sv` while the name used in the code is `new_SV`.

You've also put the name in embed.fnc within a `#if defined(PERL_IN_SV_C)` block, so the declaration isn't visible to most code, which together with the name mismatch resulted in:

```
cc -c -DPERL_CORE -D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -std=c99 -O2 -Wall -Werror=pointer-arith -Werror=vla -Wextra -Wno-long-long -Wno-declaration-after-statement -Wc++-compat -Wwrite-strings sv.c
In file included from perl.h:6186,
                 from sv.c:32:
proto.h:9141:1: warning: ‘Perl_new_sv’ declared ‘static’ but never defined [-Wunused-function]
 9141 | Perl_new_sv(pTHX_ const char *file, int line, const char *func);
      | ^~~~~~~~~~~
```

You'll probably need to add an `o` flag in embed.fnc to prevent the default new_SV() macro that will expect the file, line and function parameters.

--- a/embed.fnc
+++ b/embed.fnc
@@ -2248,6 +2248,10 @@ AMbdp	|CV *	|newSUB 	|I32 floor				\
 				|NULLOK OP *proto			\
 				|NULLOK OP *block
 ARdp	|SV *	|newSV		|const STRLEN len
+: Perl_new_sv is inline, so needs to be visible outside of sv.c
+Ciop	|SV *	|new_sv 	|NN const char *file			\
+				|int line				\
+				|NN const char *func
 Rp	|SV *	|newSVavdefelem |NN AV *av				\
 				|SSize_t ix				\
 				|bool extendible

--- a/proto.h
+++ b/proto.h
@@ -9851,6 +9851,11 @@ Perl_newSV_type_mortal(pTHX_ const svtype type)
         __attribute__always_inline__;
 # define PERL_ARGS_ASSERT_NEWSV_TYPE_MORTAL
 
+PERL_STATIC_INLINE SV *
+Perl_new_sv(pTHX_ const char *file, int line, const char *func);
+# define PERL_ARGS_ASSERT_NEW_SV                \
+        assert(file); assert(func)
+
 PERL_STATIC_INLINE void
 Perl_pop_stackinfo(pTHX);
 # define PERL_ARGS_ASSERT_POP_STACKINFO

--- a/sv.h
+++ b/sv.h
@@ -193,7 +193,7 @@ typedef enum {
 #  define HVAUX_ARENA_ROOT_IX   SVt_IV
 #endif
 #ifdef PERL_IN_SV_C
-#  define SVt_FIRST SVt_NULL	/* the type of SV that new_SV() in sv.c returns */
+#  define SVt_FIRST SVt_NULL	/* the type of SV that new_SV() in sv_inline.h returns */
 #endif
 
 #define PERL_ARENA_ROOTS_SIZE	(SVt_LAST)

--- a/sv_inline.h
+++ b/sv_inline.h
@@ -68,13 +68,15 @@
 SV* Perl_more_sv(pTHX);
 
 /* new_SV(): return a new, empty SV head */
-
-#ifdef DEBUG_LEAKING_SCALARS
-/* provide a real function for a debugger to play with */
-STATIC SV*
-S_new_SV(pTHX_ const char *file, int line, const char *func)
+PERL_STATIC_INLINE SV*
+Perl_new_sv(pTHX_ const char *file, int line, const char *func)
 {
     SV* sv;
+#ifndef DEBUG_LEAKING_SCALARS
+    PERL_UNUSED_ARG(file);
+    PERL_UNUSED_ARG(line);
+    PERL_UNUSED_ARG(func);
+#endif
 
     if (PL_sv_root)
         uproot_SV(sv);
@@ -83,6 +85,7 @@ S_new_SV(pTHX_ const char *file, int line, const char *func)
     SvANY(sv) = 0;
     SvREFCNT(sv) = 1;
     SvFLAGS(sv) = 0;
+#ifdef DEBUG_LEAKING_SCALARS
     sv->sv_debug_optype = PL_op ? PL_op->op_type : 0;
     sv->sv_debug_line = (U16) (PL_parser && PL_parser->copline != NOLINE
                 ? PL_parser->copline
@@ -99,25 +102,10 @@ S_new_SV(pTHX_ const char *file, int line, const char *func)
     MEM_LOG_NEW_SV(sv, file, line, func);
     DEBUG_m(PerlIO_printf(Perl_debug_log, "0x%" UVxf ": (%05ld) new_SV (from %s:%d [%s])\n",
             PTR2UV(sv), (long)sv->sv_debug_serial, file, line, func));
-
+#endif
     return sv;
 }
-#  define new_SV(p) (p)=S_new_SV(aTHX_ __FILE__, __LINE__, FUNCTION__)
-
-#else
-#  define new_SV(p) \
-    STMT_START {                                       \
-        if (PL_sv_root)                                        \
-            uproot_SV(p);                              \
-        else                                           \
-            (p) = Perl_more_sv(aTHX);                     \
-        SvANY(p) = 0;                                  \
-        SvREFCNT(p) = 1;                               \
-        SvFLAGS(p) = 0;                                        \
-        MEM_LOG_NEW_SV(p, __FILE__, __LINE__, FUNCTION__);  \
-    } STMT_END
-#endif
-
+#  define new_SV(p) (p)=Perl_new_sv(aTHX_ __FILE__, __LINE__, FUNCTION__)
 
 typedef struct xpvhv_with_aux XPVHV_WITH_AUX;
 

